src/apiClient.ts

```diff
@@ -42,9 +49,22 @@ class ApiClient {
     })
   }
 
-  public setTokens(tokens: Tokens): void {
-    this.tokens.accessToken = tokens.accessToken
-    this.tokens.refreshToken = tokens.refreshToken
+  public getConfiguration() {
+    return {
+      ...this.configuration,
+      configured: Object.keys(this.configuration).length > 0,
+      version: this.version,
+    }
+  }
+
+  public setAccessToken(accessToken: string): void {
+    this.accessToken = accessToken
+  }
+
+  public setTokenRefreshHandler(
```

Comment:
what's the motivation to have `setTokenRefreshHandler` as a function on the singleton over passing it with the configuration initially?

Reply:
My though behind was that the configuration is something that can be serialized to a static file eg. loading it form the .env files where as the handler needs to be passed as function that may need application context...
If you check the implementation in DH https://github.com/carforyou/carforyou-dealerhub-web/pull/744/files#diff-bc6a44fc684b34709ef25152c25c9b15R9 the configuration can just be "statically" loaded on the module top level where as the refresh handler needs to be set in the middleware to have the proper context